src/ax_interface/mib.py

```diff
@@ -127,10 +135,16 @@ def __init__(self, subtree, sub_ids, value_type, callable_, *args, updater=None)
 
     def __iter__(self):
         for sub_id in self.sub_ids:
-            yield self.sub_id + (sub_id,)
+            if isinstance(sub_id, tuple):
```

Comment:
Consider moving these if/else statements out of ContextualMIBEntry into their own class (avoiding any changes to CME)

```python
class MIBSubtreeEntry(MIBEntry):
  def __init__(...):
    # impl
  def __iter__(self):
    # impl
  def __call__(self, sub_id_group)
   # impl
``` 

and remember to replace the type inspection for `ContextualMIBEntry` in the metaclass `__new__` with `MIBSubtreeEntry`